components/brave_sync/client/bookmark_change_processor.cc

```diff
@@ -935,15 +989,18 @@ void BookmarkChangeProcessor::SendUnsynced(
 
       std::string last_send_time;
       node->GetMetaInfo("last_send_time", &last_send_time);
+      size_t current_retry_number = GetCurrentRetryNumber(node, last_send_time);
       if (!last_send_time.empty() &&
           // don't send more often than unsynced_send_interval_
           (base::Time::Now() -
               base::Time::FromJsTime(std::stod(last_send_time))) <
-          unsynced_send_interval)
+          GetRetryExponentialWaitAmount(current_retry_number)) {
         continue;
+      }
 
       bookmark_model_->SetNodeMetaInfo(node,
           "last_send_time", std::to_string(base::Time::Now().ToJsTime()));
+      SetCurrentRetryNumber(bookmark_model_, node, current_retry_number + 1);
```

Comment:
why do we count normal sending as retry?

Reply:
We count normal sending as retry to unify code of normal sending and retry by assuming normal sending is a retry with 0 resend period.
This costs just one extra 0 in the beginning of the intervals array.

Reply:
again, normal sending will not check `GetRetryExponentialWaitAmount` because its last_send_time is empty

Reply:
Fixed by https://github.com/brave/brave-core/pull/2986/commits/b4b06a795c5c846669c43aa77a993e06fa4cc2e1 and https://github.com/brave/brave-core/pull/2986/commits/46aaf45f233ed28873a63e233e552cb52dc9b60b